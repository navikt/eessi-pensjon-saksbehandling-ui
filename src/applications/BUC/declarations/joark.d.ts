
export interface JoarkFileVariant {
  variantformat:string;
  filnavn: string;
}

export interface JoarkDoc {
  dokumentInfoId: string;
  tittel: string;
  dokumentvarianter: Array<JoarkFileVariant>;
}
export interface JoarkPoster {
  journalpostId: string;
  tittel: string;
  tema: string;
  datoOpprettet: string;
  tilleggsopplysninger?: string;
  dokumenter: Array<JoarkDoc>
}

export interface JoarkFile {
  journalpostId: string;
  tittel: string;
  tema: string;
  datoOpprettet: Date;
  dokumentInfoId: string;
  tilleggsopplysninger?: string;
  variant: JoarkFileVariant
}

interface Content {
  base64: string;
}

export interface JoarkFileWithContent extends JoarkFile {
  content: Content;
  name: string;
  size: number;
  mimetype: string;
}
