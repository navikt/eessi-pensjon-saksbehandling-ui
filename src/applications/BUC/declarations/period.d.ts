export interface PeriodDate {
  day: string;
  month: string;
  year: string;
}

export interface Country {
  label: string;
  value: string;
}

export interface Period {
  id?: number,
  type: string;
  startDate: PeriodDate;
  endDate?: PeriodDate;
  dateType: string;
  uncertainDate: boolean;
  country: Country;
  comment: string;
  otherType?: string;
  insuranceName?: string;
  insuranceType?: string;
  insuranceId ?: string;
  workActivity ?: string;
  workName ?: string;
  workType ?: string;
  workStreet ?: string;
  workCity ?: string;
  workZipCode ?: string;
  workRegion ?: string;
  childFirstName ?: string;
  childLastName ?: string;
  childBirthDate ?: PeriodDate;
  learnInstitution ?: string;
  payingInstitution ?: string;
  attachments ?: Array[any];
  [key: string]: any;
}

export type Periods = Array<Period>

export type PeriodErrors = {[k: string]: string | undefined}

export interface PayloadPeriod {
  land: string;
  periode: null | {
    lukketPeriode?: {
      fom: string | null;
      tom: string | null;
    }
    openPeriode?:  {
      fom: string | null;
      extra: string;
    }
  };
  vedlegg ?: Array[any];
  trygdeordningnavn ?: string;
  medlemskap ?: string;
  forsikkringEllerRegistreringNr ?: string;
  annenInformasjon ?: string;
  usikkerDatoIndikator ?: string;
  jobbUnderAnsattEllerSelvstendig ?: string;
  navnFirma ?: string;
  typePeriode ?: string;
  adresseFirma ?: {
    postnummer: string | undefined;
    by: string | undefined;
    land: string;
    gate: string | undefined;
    region: string | undefined;
  };
  navnPaaInstitusjon ?: string;
  informasjonBarn ?:  {
    etternavn: string | null | undefined;
    fornavn: string | null | undefined;
    foedseldato: string | null | undefined;
    land: string;
  }
}

export interface Payload {
  ansattSelvstendigPerioder ?: Array<PayloadPeriod>;
  boPerioder ?: Array<PayloadPeriod>;
  barnepassPerioder ?: Array<PayloadPeriod>;
  frivilligPerioder ?: Array<PayloadPeriod>;
  forsvartjenestePerioder ?: Array<PayloadPeriod>;
  foedselspermisjonPerioder ?: Array<PayloadPeriod>;
  opplaeringPerioder ?: Array<PayloadPeriod>;
  arbeidsledigPerioder ?: Array<PayloadPeriod>;
  sykePerioder ?: Array<PayloadPeriod>;
  andrePerioder ?: Array<PayloadPeriod>;
}

export interface P4000Info {
  person: any,
  bank: any,
  stayAbroad: Array<Period>
}

export interface P4000PayloadInfo {
  personInfo: any,
  bankInfo: any,
  periodeInfo: Array<PayloadPeriod>
}
