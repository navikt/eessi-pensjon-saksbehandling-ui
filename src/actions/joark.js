import _ from 'lodash'
import * as api from 'eessi-pensjon-ui/dist/api'
import * as types from 'constants/actionTypes'
import { IS_TEST } from 'constants/environment'
import * as urls from 'constants/urls'
import sampleJoark from 'resources/tests/sampleJoark'
const sprintf = require('sprintf-js').sprintf

export const listJoarkFiles = (userId) => {
  return api.call({
    url: sprintf(urls.API_JOARK_LIST_URL, { userId: userId }),
    expectedPayload: sampleJoark.mockdata,
    type: {
      request: types.JOARK_LIST_REQUEST,
      success: types.JOARK_LIST_SUCCESS,
      failure: types.JOARK_LIST_FAILURE
    }
  })
}

export const getPreviewJoarkFile = (item) => {
  return api.call({
    url: sprintf(urls.API_JOARK_GET_URL, {
      dokumentInfoId: item.dokumentInfoId,
      journalpostId: item.journalpostId,
      variantformat: item.variant.variantformat
    }),
    expectedPayload: getMockedPayload(item.journalpostId),
    context: item,
    type: {
      request: types.JOARK_PREVIEW_REQUEST,
      success: types.JOARK_PREVIEW_SUCCESS,
      failure: types.JOARK_PREVIEW_FAILURE
    }
  })
}

export const setPreviewJoarkFile = (item) => {
  return {
    type: types.JOARK_PREVIEW_SET,
    payload: item
  }
}

export const getJoarkFile = (item) => {
  return api.call({
    url: sprintf(urls.API_JOARK_GET_URL, {
      dokumentInfoId: item.dokumentInfoId,
      journalpostId: item.journalpostId,
      variantformat: item.variant.variantformat
    }),
    expectedPayload: getMockedPayload(item.journalpostId),
    context: item,
    type: {
      request: types.JOARK_GET_REQUEST,
      success: types.JOARK_GET_SUCCESS,
      failure: types.JOARK_GET_FAILURE
    }
  })
}

export const getMockedPayload = (journalpostId) => {
  if (urls.HOST === 'localhost' && !IS_TEST) {
    const item = _.find(sampleJoark.mockdata.data.dokumentoversiktBruker.journalposter, { journalpostId: journalpostId })
    return {
      fileName: item.tittel,
      contentType: 'application/pdf',
      filInnhold: sampleJoark.files[item.tittel]
    }
  }
  return undefined
}
