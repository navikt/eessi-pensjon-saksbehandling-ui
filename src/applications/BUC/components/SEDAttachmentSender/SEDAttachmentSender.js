import React, { useEffect, useState, useCallback } from 'react'
import PT from 'prop-types'
import classNames from 'classnames'
import _ from 'lodash'
import { IS_TEST } from 'constants/environment'
import { Alert, ProgressBar } from 'eessi-pensjon-ui'

const SEDAttachmentSender = ({ allAttachments, attachmentsError, className, payload = {}, onFinished, savedAttachments, sendAttachmentToSed, t }) => {
  const [sendingAttachment, setSendingAttachment] = useState(false)
  const [_storeAttachments, setStoreAttachments] = useState(savedAttachments || [])

  const handleFinished = useCallback(() => {
    if (_(onFinished).isFunction()) {
      onFinished()
    }
  }, [onFinished])

  useEffect(() => {
    // all attachments are sent - conclude
    if (allAttachments.length === _storeAttachments.length) {
      if (!IS_TEST) {
        console.log('SEDAttachmentSender: allAttachments (' + allAttachments.length +
          ') same as _storeAttachments (' + _storeAttachments.length + ')')
      }
      handleFinished()
    }
  }, [_storeAttachments, allAttachments, handleFinished])

  useEffect(() => {
    if (!sendingAttachment) {
      if (!IS_TEST) {
        console.log('SEDAttachmentSender: Picking a new unsent attachment')
      }
      const unsentAttachments = allAttachments.filter(a => {
        return !_.find(_storeAttachments, b => {
          return a.dokumentInfoId === b.dokumentInfoId &&
            a.journalpostId === b.journalpostId &&
            a.variant.variantformat === b.variant.variantformat
        })
      })

      // all sent, mark sending attachments as done
      if (_.isEmpty(unsentAttachments)) {
        if (!IS_TEST) {
          console.log('SEDAttachmentSender: No more unsent attachment')
        }
        handleFinished()
        return
      }

      const unsentAttachment = _.first(unsentAttachments)
      const params = {
        ...payload,
        journalpostId: unsentAttachment.journalpostId,
        dokumentInfoId: unsentAttachment.dokumentInfoId,
        variantformat: unsentAttachment.variant.variantformat
      }
      if (!IS_TEST) {
        console.log('Sending unsent attachment ' + (_storeAttachments.length + 1) + ' of ' + allAttachments.length + ': ' +
          unsentAttachment.journalpostId + '/' + unsentAttachment.dokumentInfoId + '/' + unsentAttachment.variant.variantformat)
      }
      sendAttachmentToSed(params, unsentAttachment)
      setSendingAttachment(true)
    }
  }, [sendAttachmentToSed, _storeAttachments, allAttachments, handleFinished, payload, sendingAttachment])

  useEffect(() => {
    // handle if we have a newly sent attachment
    if (sendingAttachment && _storeAttachments.length !== savedAttachments.length) {
      if (!IS_TEST) {
        console.log('SEDAttachmentSender: Attachment ' + (savedAttachments.length) + ' of ' + allAttachments.length + ' sent')
      }
      setStoreAttachments(savedAttachments)
      setSendingAttachment(false)
    }
  }, [allAttachments, savedAttachments, sendingAttachment, _storeAttachments])

  const current = (_storeAttachments.length === allAttachments.length ? _storeAttachments.length : _storeAttachments.length + 1)
  const total = allAttachments.length
  const percentage = (Math.floor((current * 100) / total))

  if (attachmentsError) {
    return (
      <Alert
        type='client'
        message={t('buc:error-sendingAttachments')}
        status='ERROR'
      />
    )
  }
  return (
    <div className={classNames('a-buc-c-sedAttachmentSender', className)}>
      <ProgressBar now={percentage}>
        {t('buc:loading-sendingXofY', {
          current: current,
          total: total
        })}
      </ProgressBar>
    </div>
  )
}

SEDAttachmentSender.propType = {
  allAttachments: PT.array,
  attachmentsError: PT.boolean,
  className: PT.string,
  onFinished: PT.func,
  payload: PT.object,
  savedAttachments: PT.array,
  sendAttachmentToSed: PT.func.isRequired,
  t: PT.func.isRequired
}

export default SEDAttachmentSender
