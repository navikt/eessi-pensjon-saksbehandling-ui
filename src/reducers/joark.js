import * as types from '../constants/actionTypes'
import _ from 'lodash'

export const initialJoarkState = {
  list: undefined
}

const joarkReducer = (state = initialJoarkState, action = {}) => {
  switch (action.type) {

  case types.JOARK_LIST_SUCCESS:
    return {
      ...state,
      list: action.payload
    }

  case types.JOARK_GET_SUCCESS:
    return {
      ...state,
      file: action.payload
    }

  default:
    return state
  }
}

export default joarkReducer
