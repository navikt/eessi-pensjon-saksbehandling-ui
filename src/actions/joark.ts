import * as types from 'constants/actionTypes'
import * as urls from 'constants/urls'
import { JoarkFile } from 'declarations/joark'
import * as api from 'eessi-pensjon-ui/dist/api'
import { ActionWithPayload } from 'eessi-pensjon-ui/dist/declarations/types'
import mockJoarkRaw from 'mocks/joark/joarkRaw'
import mockJoarkPayload from 'mocks/joark/payload'

const sprintf = require('sprintf-js').sprintf

export const listJoarkFiles = (userId: string): Function => {
  return api.call({
    url: sprintf(urls.API_JOARK_LIST_URL, { userId: userId }),
    expectedPayload: mockJoarkRaw.mockdata,
    type: {
      request: types.JOARK_LIST_REQUEST,
      success: types.JOARK_LIST_SUCCESS,
      failure: types.JOARK_LIST_FAILURE
    }
  })
}

export const getPreviewJoarkFile = (item: JoarkFile): Function => {
  return api.call({
    url: sprintf(urls.API_JOARK_GET_URL, {
      dokumentInfoId: item.dokumentInfoId,
      journalpostId: item.journalpostId,
      variantformat: item.variant.variantformat
    }),
    expectedPayload: mockJoarkPayload(item.journalpostId),
    context: item,
    type: {
      request: types.JOARK_PREVIEW_REQUEST,
      success: types.JOARK_PREVIEW_SUCCESS,
      failure: types.JOARK_PREVIEW_FAILURE
    }
  })
}

export const setPreviewJoarkFile = (item: JoarkFile | undefined): ActionWithPayload<JoarkFile | undefined> => ({
  type: types.JOARK_PREVIEW_SET,
  payload: item
})
