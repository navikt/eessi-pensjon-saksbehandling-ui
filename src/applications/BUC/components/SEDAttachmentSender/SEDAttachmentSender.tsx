import classNames from 'classnames'
import { IS_TEST } from 'constants/environment'
import { JoarkFile, JoarkFiles } from 'declarations/joark'
import { JoarkFilesPropType } from 'declarations/joark.pt'
import Ui from 'eessi-pensjon-ui'
import _ from 'lodash'
import PT from 'prop-types'
import React, { useCallback, useEffect, useState } from 'react'
import { useTranslation } from 'react-i18next'

export interface SEDAttachmentPayload {
  aktoerId: string;
  rinaId: string;
  rinaDokumentId: string;
}

export interface SEDAttachmentPayloadWithFile extends SEDAttachmentPayload {
  journalpostId: string;
  dokumentInfoId: string;
  variantformat: string;
}

export interface SEDAttachmentSenderProps {
  allAttachments: JoarkFiles;
  attachmentsError ?: boolean;
  className?: string;
  initialStatus ?: string;
  onFinished ?: () => void;
  payload: SEDAttachmentPayload;
  savedAttachments: JoarkFiles;
  sendAttachmentToSed : (params: SEDAttachmentPayloadWithFile, unsent: JoarkFile) => void;
}

const SEDAttachmentSender: React.FC<SEDAttachmentSenderProps> = ({
  allAttachments, attachmentsError, className, initialStatus = 'inprogress',
  payload, onFinished, savedAttachments, sendAttachmentToSed
}: SEDAttachmentSenderProps): JSX.Element => {
  const [sendingAttachment, setSendingAttachment] = useState<boolean>(false)
  const [_storeAttachments, setStoreAttachments] = useState<JoarkFiles>(savedAttachments || [])
  const [status, setStatus] = useState<string>(initialStatus)
  const { t } = useTranslation()
  const handleFinished: Function = useCallback(() => {
    if (_.isFunction(onFinished)) {
      onFinished()
    }
  }, [onFinished])

  useEffect(() => {
    // all attachments are sent - conclude
    if (allAttachments && _storeAttachments && allAttachments.length === _storeAttachments.length) {
      /* istanbul ignore next */
      if (!IS_TEST) {
        console.log('SEDAttachmentSender: allAttachments (' + allAttachments.length +
          ') same as _storeAttachments (' + _storeAttachments.length + ')')
      }
      setStatus('done')
      handleFinished()
    }
  }, [_storeAttachments, allAttachments, handleFinished])

  useEffect(() => {
    if (!sendingAttachment && !_.isNil(allAttachments)) {
      /* istanbul ignore next */
      if (!IS_TEST) {
        console.log('SEDAttachmentSender: Picking a new unsent attachment')
      }
      const unsentAttachments: JoarkFiles = allAttachments.filter(a => {
        return !_.find(_storeAttachments, b => {
          return a.dokumentInfoId === b.dokumentInfoId &&
            a.journalpostId === b.journalpostId &&
            a.variant.variantformat === b.variant.variantformat
        })
      })

      // all sent, mark sending attachments as done
      if (_.isEmpty(unsentAttachments)) {
        /* istanbul ignore next */
        if (!IS_TEST) {
          console.log('SEDAttachmentSender: No more unsent attachment')
        }
        handleFinished()
        return
      }

      const unsentAttachment: JoarkFile | undefined = _.first(unsentAttachments)
      if (unsentAttachment) {
        const params: SEDAttachmentPayloadWithFile = {
          ...payload,
          journalpostId: unsentAttachment.journalpostId,
          dokumentInfoId: unsentAttachment.dokumentInfoId,
          variantformat: unsentAttachment.variant.variantformat
        }
        /* istanbul ignore next */
        if (!IS_TEST) {
          console.log('Sending unsent attachment ' + (_storeAttachments.length + 1) + ' of ' + allAttachments.length + ': ' +
            unsentAttachment.journalpostId + '/' + unsentAttachment.dokumentInfoId + '/' + unsentAttachment.variant.variantformat)
        }
        sendAttachmentToSed(params, unsentAttachment)
        setSendingAttachment(true)
      }
    }
  }, [sendAttachmentToSed, _storeAttachments, allAttachments, handleFinished, payload, sendingAttachment])

  useEffect(() => {
    // handle if we have a newly sent attachment
    if (sendingAttachment && _storeAttachments && savedAttachments && _storeAttachments.length !== savedAttachments.length) {
      /* istanbul ignore next */
      if (!IS_TEST) {
        console.log('SEDAttachmentSender: Attachment ' + (savedAttachments.length) + ' of ' + allAttachments.length + ' sent')
      }
      setStoreAttachments(savedAttachments)
      setSendingAttachment(false)
    }
  }, [allAttachments, savedAttachments, sendingAttachment, _storeAttachments])

  useEffect(() => {
    if (attachmentsError) {
      setStatus('error')
    }
  }, [attachmentsError, setStatus])

  if (_.isNil(_storeAttachments) || _.isNil(allAttachments)) {
    return <div />
  }

  const current: number = (_storeAttachments.length === allAttachments.length ? _storeAttachments.length : _storeAttachments.length + 1)
  const total: number = allAttachments.length
  const percentage: number = (Math.floor((current * 100) / total))

  return (
    <div className={classNames('a-buc-c-sedAttachmentSender', className)}>
      <Ui.ProgressBar now={percentage} status={status}>
        {status === 'inprogress' ? t('buc:loading-sendingXofY', {
          current: current,
          total: total
        }) : null}
        {status === 'done' ? t('buc:form-attachmentsSent') : null}
        {status === 'error' ? t('buc:error-sendingAttachments') : null}
      </Ui.ProgressBar>
    </div>
  )
}

SEDAttachmentSender.propTypes = {
  allAttachments: JoarkFilesPropType.isRequired,
  attachmentsError: PT.bool,
  className: PT.string,
  initialStatus: PT.string,
  onFinished: PT.func,
  payload: PT.any,
  savedAttachments: JoarkFilesPropType.isRequired,
  sendAttachmentToSed: PT.func.isRequired
}

export default SEDAttachmentSender
