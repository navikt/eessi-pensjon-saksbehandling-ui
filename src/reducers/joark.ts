import * as types from 'constants/actionTypes'
import { JoarkFileWithContent, JoarkPoster } from 'declarations/joark'
import { ActionWithPayload } from 'js-fetch-api'

export interface JoarkState {
  list: Array<JoarkPoster> | undefined
  previewFile: JoarkFileWithContent | undefined
}

export const initialJoarkState: JoarkState = {
  list: undefined,
  previewFile: undefined
}

const joarkReducer = (state: JoarkState = initialJoarkState, action: ActionWithPayload) => {
  switch (action.type) {
    case types.JOARK_LIST_SUCCESS:
    return {
      ...state,
      list: action.payload.data.dokumentoversiktBruker.journalposter
    }

    case types.JOARK_PREVIEW_SET:
      return {
        ...state,
        previewFile: action.payload
      }

    case types.JOARK_PREVIEW_SUCCESS:
      return {
        ...state,
        previewFile: {
          ...action.context,
          name: action.payload.fileName,
          size: action.payload.filInnhold.length,
          mimetype: action.payload.contentType,
          content: {
            base64: action.payload.filInnhold
          }
        } as JoarkFileWithContent
      }

    default:
      return state
  }
}

export default joarkReducer
