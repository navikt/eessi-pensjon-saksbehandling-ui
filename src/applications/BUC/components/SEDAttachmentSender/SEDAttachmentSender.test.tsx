import { JoarkFile } from 'applications/BUC/declarations/joark'
import { mount, ReactWrapper } from 'enzyme'
import React from 'react'
import sampleJoarkReduced from 'resources/tests/sampleJoarkReduced'
import SEDAttachmentSender, { SEDAttachmentSenderProps } from './SEDAttachmentSender'

describe('applications/BUC/components/SEDAttachmentSender/SEDAttachmentSender', () => {
  let wrapper: ReactWrapper
  const initialMockProps: SEDAttachmentSenderProps = {
    allAttachments: sampleJoarkReduced as Array<JoarkFile>,
    attachmentsError: false,
    className: 'mock-sedAttachmentSender',
    initialStatus: 'inprogress',
    onFinished: jest.fn(),
    payload: {},
    savedAttachments: [],
    sendAttachmentToSed: jest.fn(),
    t: jest.fn(t => t)
  }

  beforeEach(() => {
    wrapper = mount(<SEDAttachmentSender {...initialMockProps} />)
  })

  afterEach(() => {
    wrapper.unmount()
  })

  it('Renders', () => {
    expect(wrapper.isEmptyRender()).toBeFalsy()
    expect(wrapper).toMatchSnapshot()
  })

  it('Has proper HTML structure', () => {
    expect(wrapper.exists('.a-buc-c-sedAttachmentSender')).toBeTruthy()
  })
})
