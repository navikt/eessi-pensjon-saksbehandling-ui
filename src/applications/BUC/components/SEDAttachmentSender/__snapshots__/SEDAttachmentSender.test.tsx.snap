// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`applications/BUC/components/SEDAttachmentSender/SEDAttachmentSender Renders 1`] = `
<SEDAttachmentSender
  allAttachments={
    Array [
      Object {
        "datoOpprettet": 2018-12-27T13:42:24.000Z,
        "dokumentInfoId": "4",
        "journalpostId": "1",
        "tema": "foo",
        "tittel": "blue.pdf",
        "variant": Object {
          "filnavn": "23534345.pdf",
          "variantformat": "ARKIV",
        },
      },
      Object {
        "datoOpprettet": 2018-12-27T13:42:24.000Z,
        "dokumentInfoId": "4",
        "journalpostId": "1",
        "tema": "foo",
        "tittel": "blue.pdf",
        "variant": Object {
          "filnavn": "908745345.pdf",
          "variantformat": "DUMMY",
        },
      },
      Object {
        "datoOpprettet": 2018-12-17T13:42:24.000Z,
        "dokumentInfoId": "5",
        "journalpostId": "2",
        "tema": "bar",
        "tittel": "red.pdf",
        "variant": Object {
          "filnavn": "98734213487.pdf",
          "variantformat": "ARKIV",
        },
      },
      Object {
        "datoOpprettet": 2018-12-17T13:42:24.000Z,
        "dokumentInfoId": "5",
        "journalpostId": "2",
        "tema": "bar",
        "tittel": "red.pdf",
        "variant": Object {
          "filnavn": "998424576.pdf",
          "variantformat": "DUMMY",
        },
      },
      Object {
        "datoOpprettet": 2018-12-07T13:42:24.000Z,
        "dokumentInfoId": "6",
        "journalpostId": "3",
        "tema": "fuzz",
        "tittel": "yellow.pdf",
        "variant": Object {
          "filnavn": "1078475197.pdf",
          "variantformat": "ARKIV",
        },
      },
      Object {
        "datoOpprettet": 2018-12-07T13:42:24.000Z,
        "dokumentInfoId": "6",
        "journalpostId": "3",
        "tema": "fuzz",
        "tittel": "yellow.pdf",
        "variant": Object {
          "filnavn": "09873539824762.pdf",
          "variantformat": "DUMMY",
        },
      },
    ]
  }
  attachmentsError={false}
  className="mock-sedAttachmentSender"
  initialStatus="inprogress"
  onFinished={[MockFunction]}
  payload={
    Object {
      "aktoerId": "123",
      "rinaDokumentId": "789",
      "rinaId": "456",
    }
  }
  savedAttachments={Array []}
  sendAttachmentToSed={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "aktoerId": "123",
            "dokumentInfoId": "4",
            "journalpostId": "1",
            "rinaDokumentId": "789",
            "rinaId": "456",
            "variantformat": "ARKIV",
          },
          Object {
            "datoOpprettet": 2018-12-27T13:42:24.000Z,
            "dokumentInfoId": "4",
            "journalpostId": "1",
            "tema": "foo",
            "tittel": "blue.pdf",
            "variant": Object {
              "filnavn": "23534345.pdf",
              "variantformat": "ARKIV",
            },
          },
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <div
    className="a-buc-c-sedAttachmentSender mock-sedAttachmentSender"
  >
    <Component
      now={16}
      status="inprogress"
    >
      <div
        className="c-progressbar w-100"
      >
        <div
          aria-valuemax={100}
          aria-valuemin={0}
          aria-valuenow={16}
          className="c-progressbar__bar c-progressbar__bar-inprogress"
          role="progressbar"
          style={
            Object {
              "width": "16%",
            }
          }
        >
          buc:loading-sendingXofY
        </div>
      </div>
    </Component>
  </div>
</SEDAttachmentSender>
`;
